### PR TITLE
Add API to undelete delete group shares

### PR DESCRIPTION
**Pull request by rullzer**
When a group share is deleted we keep track of this in the DB.
Right now it is only possible for a recipient to get back the share by
asking the sharer to delete it and to share it again. This doesn't
scale.

This endpoint makes it possible to get back the share.

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

File: lib/public/Share/IManager.php [hunk 1]
```diff
@@ -134,6 +149,20 @@ public function getSharesBy($userId, $shareType, $path = null, $reshares = false
...
+	 * Get deleted shares shared with $user.
+	 * Filter by $node if provided
+	 *
+	 * @param string $userId
+	 * @param int $shareType
+	 * @param Node|null $node
+	 * @param int $limit The maximum number of shares returned, -1 for all
+	 * @param int $offset
+	 * @return IShare[]
+	 * @since 9.0.0
```
Comment (MorrisJobke) [inline lib/public/Share/IManager.php]: Let me fix this. ;)

**Review by MorrisJobke — APPROVED** [review]
Tested and code looks good 👍

**Comment by MorrisJobke** [comment]
@rullzer Could you fix the CI - then this is good to go in.

**Comment by skjnldsv** [comment]
@rullzer :see_no_evil:

File: apps/files_sharing/lib/Controller/DeletedShareAPIController.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace OCA\Files_Sharing\Controller;
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/DeletedShareAPIController.php]: license missing

**Comment by rullzer** [comment]
@skjnldsv could you have a look at the failing js unit tests?

**Comment by skjnldsv** [comment]
@rullzer sorry, missed this
will fix asap

**Comment by MorrisJobke** [comment]
* [ ] JS unit tests do still fail:

```
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action clicking share action opens sidebar and share tab FAILED
	Expected false to equal true.
	apps/files_sharing/tests/js/shareSpec.js:251:50
	TypeError: null is not an object (evaluating 'showDetailsViewStub.getCall(0).args') in apps/files_sharing/tests/js/shareSpec.js (line 252)
	apps/files_sharing/tests/js/shareSpec.js:252:41
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action adds share icon after sharing a non-shared file FAILED
	TypeError: undefined is not an object (evaluating 'shareTab._dialog.model') in apps/files_sharing/tests/js/shareSpec.js (line 276)
	apps/files_sharing/tests/js/shareSpec.js:276:20
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action updates share icon after updating shares of a file FAILED
	TypeError: undefined is not an object (evaluating 'shareTab._dialog.model') in apps/files_sharing/tests/js/shareSpec.js (line 308)
	apps/files_sharing/tests/js/shareSpec.js:308:20
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action removes share icon after removing all shares from a file FAILED
	TypeError: undefined is not an object (evaluating 'shareTab._dialog.model') in apps/files_sharing/tests/js/shareSpec.js (line 340)
	apps/files_sharing/tests/js/shareSpec.js:340:20
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action keep share text after updating reshare FAILED
	TypeError: undefined is not an object (evaluating 'shareTab._dialog.model') in apps/files_sharing/tests/js/shareSpec.js (line 367)
	apps/files_sharing/tests/js/shareSpec.js:367:20
PhantomJS 2.1.1 (Linux 0.0.0) OCA.Sharing.Util tests Share action keep share text after unsharing reshare FAILED
	TypeError: undefined is not an object (evaluating 'shareTab._dialog.model') in apps/files_sharing/tests/js/shareSpec.js (line 398)
	apps/files_sharing/tests/js/shareSpec.js:398:20
```

**Comment by skjnldsv** [comment]
Failure unrelated:
`sh: 1: kill: No such process
`

**Pull request merged**